StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1028CodeFixProvider.cs

```diff
@@ -0,0 +1,99 @@
+                var trivia = root.FindTrivia(diagnosticSpan.Start, findInsideTrivia: true);
+                if (trivia.Kind() != SyntaxKind.None) // exclude XmlText node till we have support for it.
+                {
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            "Remove trailing whitespace",
+                            ct => RemoveWhitespaceAsync(context.Document, diagnostic, ct)),
+                        diagnostic);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Removes trailing whitespace.
+        /// </summary>
+        /// <param name="document">The document to be changed.</param>
+        /// <param name="diagnostic">The diagnostic to fix.</param>
+        /// <param name="cancellationToken">The cancellation token associated with the fix action.</param>
+        /// <returns>The transformed document.</returns>
+        private static async Task<Document> RemoveWhitespaceAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken);
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+            var trivia = root.FindTrivia(diagnosticSpan.Start, findInsideTrivia: true);
+            SyntaxNode newRoot;
+            switch (trivia.Kind())
+            {
+                case SyntaxKind.WhitespaceTrivia:
+                    newRoot = root.ReplaceTrivia(trivia, SyntaxTriviaList.Empty);
+                    break;
+                case SyntaxKind.SingleLineCommentTrivia:
+                    string newTriviaContent = trivia.ToFullString().Substring(0, trivia.FullSpan.Length - diagnosticSpan.Length);
+                    var newTrivia = SyntaxFactory.SyntaxTrivia(SyntaxKind.SingleLineCommentTrivia, newTriviaContent);
+                    newRoot = root.ReplaceTrivia(trivia, newTrivia);
+                    break;
+                case SyntaxKind.MultiLineCommentTrivia:
+                case SyntaxKind.MultiLineDocumentationCommentTrivia:
+                    var triviaLocation = trivia.GetLocation();
+                    string oldTriviaContent = trivia.ToFullString();
+                    TextSpan diagnosticSpanWithinTrivia = TextSpan.FromBounds(diagnosticSpan.Start - triviaLocation.SourceSpan.Start, diagnosticSpan.End - triviaLocation.SourceSpan.Start);
+                    newTriviaContent = string.Concat(
+                        oldTriviaContent.Substring(0, diagnosticSpanWithinTrivia.Start),
+                        oldTriviaContent.Substring(diagnosticSpanWithinTrivia.End));
+                    newTrivia = SyntaxFactory.SyntaxTrivia(trivia.Kind(), newTriviaContent);
+                    newRoot = root.ReplaceTrivia(trivia, newTrivia);
+                    break;
+                default:
+                    Debug.Assert(false, "Unexpected this is.");
+                    newRoot = root;
```

Comment:
:bulb: I've seen every manner of horrible handling on the IDE side for this. It may have been fixed, but the sure way to not have it go wrong is to just `return document;` (the input unaltered).